### ISSUE TITLE
Keyword analysis for EDN

### ISSUE THREAD

**Issue by elken** [clj-kondo/clj-kondo#1926]
To upvote this issue, give it a thumbs up. See [this list](https://github.com/clj-kondo/clj-kondo/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc) for the most upvoted issues.

[ To keep development of this project going, consider sponsoring. If you are
already a sponsor, thank you! You can remove this message. ]

**Is your feature request related to a problem? Please describe.**

Implementing this allows it to be possible for clojure-lsp to provide things like outlines and references for EDN files.

**Describe the solution you'd like**

I'm not sure, I'm not familiar with kondo's implementation :sweat_smile: 

**Describe alternatives you've considered**

I started writing my own minor mode to try and handle this with regexp, but it seemed more appropriate to have this implemented upstream

**Additional context**

There seems to be some overlap with https://github.com/clj-kondo/clj-kondo/issues/1902, which could bridge that gap

I am fine to try and look into adding this myself, if necessary.

**Comment by borkdude** [comment]
- Please make the examples self-contained so the person implementing this feature has to spend less time digging up what you were actually asking for
- "dependencies listed": these aren't actual keywords are they?

**Comment by elken** [comment]
> * Please make the examples self-contained so the person implementing this feature has to spend less time digging up what you were actually asking for

Sure, I'm editing it up now

>     * "dependencies listed": these aren't actual keywords are they?

Maybe "structural analysis" would be more accurate, they're still keys and still useful to navigate to

**Comment by ericdallo** [comment]
WDYT about the examples @borkdude ?
From LSP point of view, having those keywords in the `:keywords` bucket would help a lot implement find-references of those keywords and things like that

**Comment by borkdude** [comment]
Yes, this would work for `:paths` but not for symbols in `:deps`

**Comment by ericdallo** [comment]
Yes, IMO keywords is already a really good improvement as a first deliverable, we could start a new issue for symbols as it's more complex (we don't have symbol analysis ATM even on clj files), what you say?

**Comment by borkdude** [comment]
Yep, agreed

**Comment by ericdallo** [comment]
Proposal [here](https://github.com/clj-kondo/clj-kondo/pull/1934), clojure-lsp WIP branch with working find-references of keywords in edn [here](https://github.com/clojure-lsp/clojure-lsp/pull/1441) in case anyone wanna try

**Comment by elken** [comment]
> Proposal [here](https://github.com/clj-kondo/clj-kondo/pull/1934), clojure-lsp WIP branch with working find-references of keywords in edn [here](https://github.com/clojure-lsp/clojure-lsp/pull/1441) in case anyone wanna try

I'll give it a whirl today :) Thanks for a swift resolution!

**Comment by elken** [comment]
Added in https://github.com/clj-kondo/clj-kondo/commit/8ddf1e6c568b04c60bed29e6e3a5c05d286dbdb8
